test/common.py

```diff
@@ -0,0 +1,259 @@
+import json
+import logging
+import time
+import LogEntry_pb2
+import subprocess
+import struct
+import requests
+import os
+import requests_unixsocket
+from requests.packages.urllib3.util.retry import Retry
+from requests.adapters import HTTPAdapter
+
+logger = logging.getLogger(__name__)
+TIMEROUT = 500
+SOCKET_START_URL = "http+unix://%2Frun%2Fdocker%2Fplugins%2Fsplunklog.sock/LogDriver.StartLogging"
+SOCKET_STOP_URL = "http+unix://%2Frun%2Fdocker%2Fplugins%2Fsplunklog.sock/LogDriver.StopLogging"
+
+
+def start_logging_plugin(plugin_path):
+    '''Start to run logging plugin binary'''
+    args = (plugin_path)
+    popen = subprocess.Popen(args, stdout=subprocess.PIPE)
+
+    return popen
+
+
+def kill_logging_plugin(plugin_path):
+    '''Kill running logging plugin process'''
+    os.system("killall " + plugin_path)
+
+
+def open_fifo(fifo_location):
+    '''create and open a file'''
+    fifo_writer = open(fifo_location, 'wb')
+
+    return fifo_writer
+
+
+def write_proto_buf_message(fifo_writer=None,
+                            source="test",
+                            time_nano=int(time.time() * 1000000000),
+                            message="",
+                            partial=False,
+                            id=""):
+    '''
+    write to fifo in the format of LogMessage protobuf
+    '''
+    log = LogEntry_pb2.LogEntry()
+    log.source = source
+    log.time_nano = time_nano
+    log.line = bytes(message)
+    log.partial = partial
+
+    buf = log.SerializeToString(log)
+    size = len(buf)
+
+    size_buffer = bytearray(4)
+    struct.pack_into(">i", size_buffer, 0, size)
+    fifo_writer.write(size_buffer)
+    fifo_writer.write(buf)
+
+    fifo_writer.flush()
+    fifo_writer.close()
+
+
+def request_start_logging(file_path):
+    '''
+    send a request to the plugin to start logging
+    @param: file_path
+    '''
+    req_obj = {
+        "File": file_path,
+        "Info": {
+            "ContainerID": "test",
+            "Config": {
+                "splunk-url": "https://52.53.254.149:8088",
```

Comment:
Can we please remove the IP and token values and update them with sudo values?

Reply:
good call. added to command opts